samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,501 @@
+    return sqrt(sz2);
+}
+
+template<>
+Mat sizeCal(const Mat &w, const Mat &h)
+{
+    Mat pad = (w + h)*0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+int main(int argc, char **argv)
+{
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
+
+   if (argc == 1 || parser.has("help"))
+   {
+       parser.printMessage();
+       return 0;
+   }
+
+    std::string input = parser.get<String>("input");
+    std::string net = parser.get<String>("net");
+    std::string kernel_cls1 = parser.get<String>("kernel_cls1");
+    std::string kernel_r1 = parser.get<String>("kernel_r1");
+
+    // Read nets.
+    Net siamRPN, siamKernelCL1, siamKernelR1;
+    try
+    {
+        siamRPN = readNet(net);
+        siamKernelCL1 = readNet(kernel_cls1);
+        siamKernelR1 = readNet(kernel_r1);
+    }
+    catch (cv::Exception &ee)
+    {
+        std::cerr << "Exception: " << ee.what() << std::endl;
+        std::cout << "Can't load the network by using the flowing files:" << std::endl;
+        std::cout << "siamRPN : " << net << std::endl;
+        std::cout << "siamKernelCL1 : " << kernel_cls1 << std::endl;
+        std::cout << "siamKernelR1 : " << kernel_r1 << std::endl;
+        return 1;
+    }
+
+    const std::string preWinName = "DaSiamRPN";
+    namedWindow(preWinName, WINDOW_AUTOSIZE);
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+    bool openSuccess = parser.has("input") ? cap.open(parser.get<String>("input")) : cap.open(0);
+    CV_Assert(openSuccess);
+
+    // Read the first image.
+    cap >> image;
+    if (image.empty())
+    {
+        std::cout << "Video reading error!" << net << std::endl;
+        return -1;
+    }
+
+    std::string infor = "Select initial bounding box you want to track.";
+    std::string infor2 = " After that, Press the ENTER key.";
+    std::cout<<infor<<std::endl;
+    std::cout<<infor2<<std::endl;
+
+    putText(image, infor, Point(0, 15), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+    putText(image, infor2, Point(0, 35), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+
+    setMouseCallback(preWinName, onMouseRectPicking,0);// Use mouse to select the initial bounding box.
+
+    trackerConfig trackState;
+    trackState.update_scoreSize();
+
+    Mat tempImg;
+    char c;
+
+    while(1)
+    {
+        tempImg = image.clone();
+        trackState.targetBox = Box{float(selectRect.x) + float(selectRect.width)/2, float(selectRect.y) + float(selectRect.height)/2, float(selectRect.width), float(selectRect.height)};
+        rectangle(tempImg, selectRect,Scalar(255,0,0),2,8,0);
+
+        imshow(preWinName, tempImg);
+
+        c=(char)waitKey(30);
+        if(13==c) //Enter key
+            break;
+    }
+
+    // Set tracking template.
+    trackerInit(image, trackState, siamRPN, siamKernelR1, siamKernelCL1);
+
+    TickMeter tickMeter;
+
+    int count = 0;
+    while (waitKey(1) < 0)
```

Comment:
Only one `waitKey()` call should be per loop cycle.

Add pressed keys handling, like 27 (Esc).

Refer to code of other OpenCV samples.

Reply:
Thanks, I have fixed this.